src/lib/holocene/toggle-button/toggle-button.svelte

```diff
@@ -1,61 +1,74 @@
 <script lang="ts">
+  import type {
+    HTMLAnchorAttributes,
+    HTMLButtonAttributes,
+  } from 'svelte/elements';
+
   import { page } from '$app/stores';
-  
+
   import Icon from '$lib/holocene/icon/icon.svelte';
-import type { IconName } from '$lib/holocene/icon/paths';
-import { getAppContext } from '$lib/utilities/get-context';
+  import type { IconName } from '$lib/holocene/icon/paths';
+  import { getAppContext } from '$lib/utilities/get-context';
+
+  type BaseProps = {
+    icon?: IconName;
+    group?: boolean;
+    active?: boolean;
+    'data-testid'?: string;
+  };
+
+  type AnchorProps = BaseProps &
+    HTMLAnchorAttributes & {
+      href: string;
+      base?: string;
+    };
 
+  type ButtonProps = BaseProps &
+    HTMLButtonAttributes & {
+      href?: never;
+      base?: never;
+    };
+
+  type $$Props = AnchorProps | ButtonProps;
+
+  let className = '';
+  export { className as class };
   export let icon: IconName = null;
   export let group = getAppContext('group');
   export let href = '';
   export let base = href;
   export let active = false;
 </script>
 
-{#if href}
-  <a
-    class="flex items-center justify-center border border-gray-900 py-2 px-4 text-sm {$$props.class}"
-    class:rounded-lg={!group}
-    class:active={$page.url.pathname.includes(base) || active}
-    class:group
-    href={href + $page.url.search}
-    id={$$props.id}
-    {...$$props}
-    on:click
-  >
-    {#if icon}
-      <div class="flex items-center gap-2">
-        <Icon name={icon} />
-        <span class="hidden md:block"><slot /></span>
-      </div>
-    {:else}
-      <slot />
-    {/if}
-  </a>
-{:else}
-  <div
-    class="flex cursor-pointer items-center justify-center border border-gray-900 py-2 px-4 text-sm {$$props.class}"
-    class:rounded-lg={!group}
-    class:active
-    class:group
-    id={$$props.id}
-    {...$$props}
-    on:click
-  >
-    {#if icon}
-      <div class="flex items-center gap-2">
-        <Icon name={icon} />
-        <span class="hidden md:block"><slot /></span>
-      </div>
-    {:else}
-      <slot />
-    {/if}
-  </div>
-{/if}
+<svelte:element
+  this={href ? 'a' : 'button'}
+  class="toggle-button {className}"
+  class:group
+  class:active={href ? $page.url.pathname.includes(base) : active}
+  href={href ? href + $page.url.search : null}
+  class:rounded-lg={!group}
+  on:click
+  role="button"
+  tabindex="0"
+  {...$$restProps}
+>
+  {#if icon}
+    <div class="flex items-center gap-2">
+      <Icon name={icon} />
```

Comment:
One thing I've noticed with our `Button` like components is that if they have icons **and** text the screen reader reads both. Maybe we consider hiding icons from screen readers that are nested within interactive elements like buttons/toggles?

<img width="838" alt="Screenshot 2023-09-07 at 2 22 40 PM" src="https://github.com/temporalio/ui/assets/15069288/04f025e8-d23a-4be6-b7b5-11cfda346e0b">